drivers/sensors/adxl372_uorb.c

```diff
@@ -0,0 +1,609 @@
+{
+  /* If SPI bus is shared then lock and configure it */
+
+  SPI_LOCK(priv->spi, true);
+
+  SPI_SETFREQUENCY(priv->spi, ADXL372_SPI_FREQUENCY);
+  SPI_SETMODE(priv->spi, ADXL372_SPI_MODE);
+
+  /* Select the ADXL372 */
+
+  SPI_SELECT(priv->spi, SPIDEV_ACCELEROMETER(priv->devno), true);
+
+  /* Send register to read and get the next 2 bytes */
+
+  SPI_SEND(priv->spi, (regaddr << 1) | ADXL372_READ);
+  SPI_RECVBLOCK(priv->spi, regval, len);
+
+  /* Deselect the ADXL372 */
+
+  SPI_SELECT(priv->spi, SPIDEV_ACCELEROMETER(priv->devno), false);
+
+  /* Unlock bus */
+
+  SPI_LOCK(priv->spi, false);
+}
+
+/****************************************************************************
+ * Name: adxl372_data
+ ****************************************************************************/
+
+static int16_t adxl372_data(FAR uint8_t *data)
+{
+  return ((int16_t)(((data[0] << 8) | (data[1] & 0xf0)))) >> 4;
+}
+
+/****************************************************************************
+ * Name: adxl372_checkid
+ *
+ * Description:
+ *   Read and verify the ADXL372 chip ID
+ *
+ ****************************************************************************/
+
+static int adxl372_checkid(FAR struct adxl372_sensor_s *priv)
+{
+  uint8_t id = 0;
+
+  id = adxl372_getreg8(priv, ADXL372_DEVID_AD);
+  if (id != ADXL372_DEVID_AD_VALUE)
+    {
+      snerr("Wrong AD! %02x\n", id);
+      return -ENODEV;
+    }
+
+  id = adxl372_getreg8(priv, ADXL372_DEVID_MST);
+  if (id != ADXL372_DEVID_MST_VALUE)
+    {
+      snerr("Wrong MST! %02x\n", id);
+      return -ENODEV;
+    }
+
+  id = adxl372_getreg8(priv, ADXL372_PARTID);
+  if (id != ADXL372_PARTID_VALUE)
+    {
+      snerr("Wrong PARTID! %02x\n", id);
+      return -ENODEV;
+    }
+
+  return OK;
+}
+
+/****************************************************************************
+ * Name: adxl372_start
+ ****************************************************************************/
+
+static void adxl372_start(FAR struct adxl372_sensor_s *priv)
+{
+  adxl372_putreg8(priv, ADXL372_POWER_CTL,
+                  ADXL372_POWER_HPF_DISABLE |
+                  ADXL372_POWER_MODE_MEASURE);
+}
+
+/****************************************************************************
+ * Name: adxl372_stop
+ ****************************************************************************/
+
+static void adxl372_stop(FAR struct adxl372_sensor_s *priv)
+{
+  adxl372_putreg8(priv, ADXL372_POWER_CTL, 0);
+}
+
+/****************************************************************************
+ * Name: adxl372_reset
+ ****************************************************************************/
+
+static void adxl372_reset(FAR struct adxl372_sensor_s *priv)
+{
+  int wdcnt = 10;
+
+  /* Set stanby mode */
+
+  adxl372_putreg8(priv, ADXL372_POWER_CTL, 0);
+
+  /* Wait for boot to finish (15 ms error timeout) */
+
+  up_mdelay(5);
+  while (wdcnt > 0 && (0 != adxl372_getreg8(priv, ADXL372_RESET)))
+    {
+      up_mdelay(1);
```

Comment:
You are not decreasing the protection counter `wdcnt` to avoid stuck here forever